planning/behavior_path_planner_common/src/turn_signal_decider.cpp

```diff
@@ -392,6 +392,39 @@ TurnIndicatorsCommand TurnSignalDecider::resolve_turn_signal(
   return intersection_signal_info.turn_signal;
 }
 
+TurnSignalInfo TurnSignalDecider::overwrite_turn_signal(
+  const PathWithLaneId & path, const Pose & current_pose, const size_t current_seg_idx,
+  const TurnSignalInfo & original_signal, const TurnSignalInfo & new_signal,
+  const double nearest_dist_threshold, const double nearest_yaw_threshold)
+{
+  if (original_signal.turn_signal.command == TurnIndicatorsCommand::NO_COMMAND) {
+    return new_signal;
+  }
+
+  if (original_signal.turn_signal.command == TurnIndicatorsCommand::DISABLE) {
+    return new_signal;
+  }
+
+  const auto get_distance = [&](const Pose & input_point) {
+    const size_t nearest_seg_idx = motion_utils::findFirstNearestSegmentIndexWithSoftConstraints(
+      path.points, input_point, nearest_dist_threshold, nearest_yaw_threshold);
+    return motion_utils::calcSignedArcLength(
+      path.points, current_pose.position, current_seg_idx, input_point.position, nearest_seg_idx);
+  };
+
+  const auto & original_desired_end_point = original_signal.desired_end_point;
+  const auto & new_desired_start_point = new_signal.desired_start_point;
+
+  const double dist_to_original_desired_end =
+    get_distance(original_desired_end_point) - base_link2front_;
+  const double dist_to_new_desired_start = get_distance(new_desired_start_point) - base_link2front_;
```

Comment:
since both subtract `base_link2front_` from the `get_distance`, I am curious why it is not substract from within the `get_distance`'s lambda directly?

Reply:
:tashikani:

Reply:
https://github.com/autowarefoundation/autoware.universe/pull/7045/commits/1959d3fd77b0a1f2503c8aa3a7eece23dd57f4e9